index.js

```diff
@@ -158,6 +170,41 @@ async function fastifyRateLimit (fastify, settings) {
   }
 }
 
+async function addRouteThrottleHook (routeOptions, throttleOptions) {
+  const hook = 'onSend'
+  const hookHandler = throttleOnSendHandler(throttleOptions)
+  if (Array.isArray(routeOptions[hook])) {
+    routeOptions[hook].push(hookHandler)
+  } else if (typeof routeOptions[hook] === 'function') {
+    routeOptions[hook] = [routeOptions[hook], hookHandler]
+  } else {
+    routeOptions[hook] = [hookHandler]
+  }
+}
+
+function throttleOnSendHandler (throttleOpts) {
+  const bps = throttleOpts.bps
+  return function (request, reply, payload, done) {
+    if (payload && payload.pipe) {
+      const throttleStream = new ThrottleStream({ bps })
+      payload.pipe(throttleStream)
+      done(null, throttleStream)
+      return
+    } else if (Buffer.isBuffer(payload)) {
+      const throttleStream = new ThrottleStream({ bps })
+      Readable.from(payload).pipe(throttleStream)
```

Comment:
It does not have much sense to throttle in this case. Throlling is needed to limit the resource consumed by the caller. In this specific case, we are holding quite a bit of data in memory, literally increasing the resource consumed to process the request.